saveData.php

```diff
@@ -94,21 +95,22 @@
     }
     $module->query("COMMIT", []);
     REDCap::logEvent($logging_title, $logging_message, null, null, null, $pid);
-    $_SESSION['message'] = ucfirst(
+    $message = ucfirst(
             $type
         ) . " <strong>$old_var</strong> has been updated to <strong>$new_var</strong> successfully.";
-    $_SESSION['message_type'] = "success";
-    $status = "success_message";
+    $module->getMessageHandler()->addMessage($message);
+    $module->getMessageHandler()->messageType('success');
+    $module->getMessageHandler()->setPrintVariable($module->printVariableList($pid));
+    $module->getMessageHandler()->setPrintInstrument($module->printInstrumentList($pid));
 } catch (Exception $e) {
     $module->query("ROLLBACK", []);
-    $_SESSION['message'] = "Something went wrong when updating the " . ucfirst(
+    $message = "Something went wrong when updating the " . ucfirst(
             $type
         ) . " <em>$old_var</em> to <strong>$new_var</strong>";
-    $_SESSION['message_type'] = "warning_message";
+    $module->getMessageHandler()->addMessage($message);
+    $module->getMessageHandler()->messageType('danger');
     throw $e;
 }
 
-echo json_encode(array(
-                     'status' => "success"
-                 ));
+echo json_encode($module->getMessageHandler());
```

Comment:
I find this line a little confusing. It looks like json_encoding a class.  What happens if new properties and methods are added to this class, and we don't want those to be encoded? Or if those have sensitive information? I would add a new message to your message handler like `getMessageAttributes` that returns an explicitly defined array of what the front end expects.

Reply:
Created the new class. I like this approach.